hoot-core/src/main/cpp/hoot/core/criterion/HighwayCriterion.cpp

```diff
@@ -44,36 +45,35 @@ bool HighwayCriterion::isSatisfied(const ConstElementPtr& element) const
   const Tags& tags = element->getTags();
   const ElementType type = element->getElementType();
   Tags::const_iterator it = tags.find("highway");
-  QString key;
 
   // Is it a legit highway?
   if ((type == ElementType::Way || type == ElementType::Relation) &&
       it != tags.end() && it.value() != "")
   {
     result = true;
-    key = it.key();
   }
-
-  // Maybe it's a way with nothing but a time tag...
-  it = tags.find("source:datetime");
-  if (type == ElementType::Way && tags.keys().size() < 2 && it != tags.end())
+  //  Make sure it isn't a building or a part of a building
+  else if (_map && (BuildingCriterion(_map).isSatisfied(element) ||
+                    BuildingCriterion(_map).isParentABuilding(element->getElementId())))
```

Comment:
Here is where we make sure it isn't a building before accepting all ways with a date/time stamp as highways below.